src/main/java/simba/AllTasksManager.java

```diff
@@ -0,0 +1,403 @@
+            }
+            String[] subParts = parts[3].split("to:");
+
+            if (subParts.length < 2) {
+                throw new SimbaException("Corrupt event, skipping: " + taskLine);
+            }
+
+            String from = subParts[0];
+            String to = subParts[1];
+            Event event = new Event(description, from, to);
+
+            if (isCompleted) {
+                event.markAsCompleted();
+            }
+
+            return event;
+            default:
+                throw new SimbaException("Unknown task type. Skipping task " + taskType);
+        }
+
+    }
+
+    /**
+     * Handles user input for adding, listing, marking and unmarking tasks.
+     * User can enter command "bye" to exit
+     */
+    public void startTaskManagement() {
+        //Scanner to read input
+        System.out.println("Please enter the tasks you have yet to complete or type \"List\" to list pending tasks");
+        Scanner sc = new Scanner(System.in);
+
+        while (true) {
+            String input = sc.nextLine();
+
+            if (input.equalsIgnoreCase("bye")) {
+                userInterface.showExitMessage();
+                break;
+            } else if (input.equalsIgnoreCase("list")) {
+                try {
+                    listTasks();
+                } catch (SimbaException e) {
+                    userInterface.printDashedLine();
+                    System.out.println(e.getMessage());
+                    userInterface.printDashedLine();
+                }
+            } else if (input.startsWith("mark")) {
+                try {
+                    handleMark(input);
+                } catch (SimbaException e) {
+                    userInterface.printDashedLine();
+                    System.out.println(e.getMessage());
+                    userInterface.printDashedLine();
+                }
+            } else if (input.startsWith("unmark")) {
+                try {
+                    handleUnmark(input);
+                } catch (SimbaException e) {
+                    userInterface.printDashedLine();
+                    System.out.println(e.getMessage());
+                    userInterface.printDashedLine();
+                }
+            } else if (input.startsWith("delete")) {
+                try {
+                    handleDelete(input);
+                } catch (SimbaException e) {
+                    userInterface.printDashedLine();
+                    System.out.println(e.getMessage());
+                    userInterface.printDashedLine();
+                }
+            } else {
+                try {
+                    createTaskFromInput(input);
+                } catch (SimbaException e) {
+                    userInterface.printDashedLine();
+                    System.out.println(e.getMessage());
+                    userInterface.printDashedLine();
+                }
+            }
+        }
+    }
+
+}
```

Comment:
this class is particularly too long. see if you can extract related data/methods to separate classes